e2e/__tests__/selectProjects.test.ts

```diff
@@ -111,6 +111,131 @@ describe('Given a config with two named projects, first-project and second-proje
+      result = runWithJson('select-projects', [
+        '--ignoreProjects',
+        'first-project',
+      ]);
+    });
+    it('runs the tests in the second project only', () => {
+      expect(result.json).toHaveProperty('success', true);
+      expect(result.json).toHaveProperty('numTotalTests', 1);
+      expect(result.json.testResults.map(({name}) => name)).toEqual([
+        resolve(dir, '__tests__/second-project.test.js'),
+      ]);
+    });
+    it('prints that only second-project will run', () => {
+      expect(result.stderr).toMatch(/^Running one project: second-project/);
+    });
+  });
+
+  describe('when Jest is started with `--ignoreProjects second-project', () => {
+    let result: RunJestJsonResult;
+    beforeAll(() => {
+      result = runWithJson('select-projects', [
+        '--ignoreProjects',
+        'second-project',
+      ]);
+    });
+    it('runs the tests in the first project only', () => {
+      expect(result.json).toHaveProperty('success', true);
+      expect(result.json).toHaveProperty('numTotalTests', 1);
+      expect(result.json.testResults.map(({name}) => name)).toEqual([
+        resolve(dir, '__tests__/first-project.test.js'),
+      ]);
+    });
+    it('prints that only first-project will run', () => {
+      expect(result.stderr).toMatch(/^Running one project: first-project/);
+    });
+  });
+
+  describe('when Jest is started with `--ignoreProjects third-project`', () => {
+    let result: RunJestJsonResult;
+    beforeAll(() => {
+      result = runWithJson('select-projects', [
+        '--ignoreProjects',
+        'third-project',
+      ]);
+    });
+    it('runs the tests in the first and second projects', () => {
+      expect(result.json).toHaveProperty('success', true);
+      expect(result.json).toHaveProperty('numTotalTests', 2);
+      expect(result.json.testResults.map(({name}) => name).sort()).toEqual([
+        resolve(dir, '__tests__/first-project.test.js'),
+        resolve(dir, '__tests__/second-project.test.js'),
+      ]);
+    });
+    it('prints that both first-project and second-project will run', () => {
+      expect(result.stderr).toMatch(
+        /^Running 2 projects:\n- first-project\n- second-project/,
+      );
+    });
+  });
+
+  describe('when Jest is started with `--ignoreProjects first-project second-project`', () => {
+    let result: RunJestResult;
+    beforeAll(() => {
+      result = run('select-projects', [
+        '--ignoreProjects',
+        'first-project',
+        'second-project',
+      ]);
+    });
+    it('fails', () => {
+      expect(result).toHaveProperty('failed', true);
+    });
+    it.skip('prints that no project was found', () => {
+      expect(result.stdout).toMatch(
+        /^You provided values for --ignoreProjects, but no projects were found matching the selection/,
+      );
+    });
+  });
+
+  describe('when Jest is started with `--selectProjects first-project second-project --ignoreProjects first-project` ', () => {
```

Comment:
@SimenB  here is the test for using `--selectProjects` and `--ignoreProjects` together.